hare3/hare_test.go

```diff
@@ -454,7 +457,15 @@ func (cl *lockstepCluster) setup() {
 			Publish(gomock.Any(), gomock.Any(), gomock.Any()).
 			Do(func(ctx context.Context, _ string, msg []byte) error {
 				for _, other := range cl.nodes {
-					other.hare.Handler(ctx, n.peerId(), msg)
+					cl.logger.Debug("passing gosspied msg", zap.Int("from", n.i), zap.Int("to", other.i))
+					err := other.hare.Handler(ctx, n.peerId(), msg)
+					if err != nil {
+						// This is 'allowed' error as it happens in some tests
+						require.ErrorContains(cl.t, err, "dropped by graded gossip", "from = %d, to = %d", n.i, other.i)
+					} else {
+						require.NoError(cl.t, err, "from = %d, to = %d", n.i, other.i)
+					}
```

Comment:
I am unsure about this error assertion. What if a test that should not return an error here then returns "dropped by graded gossip"? Would it make sense to pass the expected error to the `setup` method to make sure we catch if behaviour changes?

Reply:
IDK it would be even more awkward. Ideally, the test in which these errors happen on purpose would have its own body and its own expectations.